examples/mem-alloc-test/src/main.rs

```diff
@@ -0,0 +1,152 @@
+use core::arch::global_asm;
+use std::{thread};
+use std::time::{Instant, Duration};
+use rand::Rng;
+use std::sync::{Mutex, Arc};
+
+
+enum SizeType {
+    LARGE_MEM,
+    MEDIUM_MEM,
+    LOW_MEM,
+}
+
+fn calculateAndPrintStat(shared_mutex_clone: Arc<Mutex<i32>>,
+                         shared_mutex_dur_clone: Arc<Mutex<u128>>,
+                         duration: Duration, tid: i32, memtypestr: &str) {
+
+    /* This function continuosly prints the average execution time of the worker
+     * thread which includes time to allocate a buffer + time to traverse it +
+     * time to free it.
+     * TODO: Currently the average is taken over all threads (large, medium and
+     * small). I will a separate average timings for small, medium and large
+     * buffer threads
+     */
+
+    let duration_ms =   duration.as_nanos();
+    let mut data = shared_mutex_clone.lock().unwrap();
+    *data += 1;
+    let mut data_d = shared_mutex_dur_clone.lock().unwrap();
+    *data_d += duration_ms;
+    let avg_exec_time = *data_d as f64 / *data as f64;
+
+    if ((*data % 10001 == 0 && memtypestr == "LOW") ||
+        (*data % 1001 == 0 && memtypestr == "MEDIUM") ||
+        (*data % 12 == 0 && memtypestr == "LARGE")) {
+        println!("Thread {} ({})records count {} avg execution time {}", tid,
+                 memtypestr, *data, avg_exec_time);
+    }
+}
+
+
+fn Get_Random_Num(start: usize, end: usize) -> usize {
+    let mut rng = rand::thread_rng();
+    let random_number: usize = rng.gen_range(start..=end);
+    return random_number;
+}
+
+fn worker_thread(tid: i32, shared_mutex_clone: Arc<Mutex<i32>>,
+                 shared_mutex_dur_clone: Arc<Mutex<u128>>, memtype: SizeType) {
+    loop {
```

Comment:
So this loop continues forever? As a future improvement, it may make sense to run for either a specified number of iterations or a specified amount of time. Also as a future improvement, it can take a long time for all of the threads to start up. So you might want to consider having all of the threads wait on something before they start doing work, and only get signalled to start once all of the threads have been created.